### PR TITLE
fix(transport-authority): Get correct requiresMileageRegistration value

### PR DESCRIPTION
**Pull request by johannaagma**
## What

* The value requiresMileageRegistration is only correct in the /currentvehicleswithmileageandinsp (not in /BasicVehicleInformation), we should start using that value everywhere, instead of hardcoding on our side what fueltypes are electric. Then when changes are made that all cars (not only electric) require mileage registration, the change only needs to be applied on SGS side.

* Also cleanup so make value is the same on our side. When using /BasicVehicleInformation we need to use make+vehcom, but when using /currentvehicles or /currentvehicleswithmileageandinsp we need to use only make (SGS has already added the vehcom to that field)

* Some cleanup/refactor when fetching currentvehicles + info in template-api-modules, when the totalCount of vehicles affects how much information is fetched

* Rename field isRequired -> requireMileage in dataSchema, since there was mismatch between the field name, which maybe(?) caused the bug where user was able to go through ownerchange without registering mileage (was not able to reproduce, but hopefully this will minimize the chance of that happening again)

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Enhanced vehicle data retrieval with improved error handling and validation for ownership checks.
	- Streamlined methods for fetching vehicle data and mileage readings, ensuring robust error handling.
	- Updated vehicle selection logic to simplify the retrieval of current vehicles.
	- Introduced new data transfer objects for vehicle mileage and inspection details, improving data handling.
	- Updated user interface to reflect new naming conventions for mileage requirement fields.

- **Bug Fixes**
	- Improved error messages for vehicle not found scenarios and ownership validation.

- **Documentation**
	- Updated naming conventions for mileage requirement fields across multiple services and schemas for consistency.

- **Chores**
	- Removed deprecated method for fetching electric fuel codes from the vehicle codetables client.

### CODE REVIEW

**Review by berglindoma13 — APPROVED** [review]

File: libs/api/domains/transport-authority/src/lib/transportAuthority.service.ts [hunk 1]
```diff
@@ -373,8 +362,6 @@ export class TransportAuthorityApi {
         color: vehicleInfo.color,
         make: `${vehicleInfo.make} ${vehicleInfo.vehcom}`,
         permno: vehicleInfo.permno,
-        requireMileage: vehicleInfo.requiresMileageRegistration,
-        mileageReading: vehicleInfo?.mileageReadings?.[0]?.mileage || '',
```
Comment (jonarnarbriem) [inline libs/api/domains/transport-authority/src/lib/transportAuthority.service.ts:377]: Bara til að staðfesta, við þurfum ekki km stöðu fyrir númeraplötur?

File: libs/application/template-api-modules/src/lib/modules/templates/transport-authority/change-co-owner-of-vehicle/change-co-owner-of-vehicle.service.ts [hunk 2]
```diff
@@ -87,78 +89,101 @@ export class ChangeCoOwnerOfVehicleService extends BaseTemplateApiService {
...
+    // Validate that user has at least 1 vehicle
+    if (!countResult) {
+      throw new TemplateApiError(
+        {
+          title: coreErrorMessages.vehiclesEmptyListOwner,
+          summary: coreErrorMessages.vehiclesEmptyListOwner,
+        },
+        400,
+      )
+    }
```
Comment (jonarnarbriem) [inline libs/application/template-api-modules/src/lib/modules/templates/transport-authority/change-co-owner-of-vehicle/change-co-owner-of-vehicle.service.ts:102]: Hvað gerist aftur ef við köstum templateApiError? villa til notanda og loggað sem error?

Þurfum að birta notanda skilaboð, en það er ekki error að einhver eigi ekki ökutæki. Ef samskiptin við SGS liggja niðri þá ætti að koma villa á það. Er ekki bara nóg að logga info? mögulega warning

File: libs/application/template-api-modules/src/lib/modules/templates/transport-authority/change-co-owner-of-vehicle/change-co-owner-of-vehicle.service.ts [hunk 3]
```diff
@@ -87,78 +89,101 @@ export class ChangeCoOwnerOfVehicleService extends BaseTemplateApiService {
...
+        // Display radio buttons, validate all vehicles now
+        return this.mapVehicle(auth, vehicle, true)
+      }),
+    )
+
+    return {
+      totalRecords: countResult,
+      vehicles: vehicles,
+    }
+  }
```
Comment (jonarnarbriem) [inline libs/application/template-api-modules/src/lib/modules/templates/transport-authority/change-co-owner-of-vehicle/change-co-owner-of-vehicle.service.ts:142]: Við erum að kalla tvisvar í currentvehicleswithmileageandinspGet, getum við ekki samnýtt þessi köll svo það sé bara eitt?

Stækkum page size í fyrra kallinu í 20 og ef það eru fleiri en 1 bls þá vitum við að viðkomandi á fleiri en 20 ökutæki.

Svo getum við notað result.data.length til að vita hvort notandi eigi fleiri eða færri en 5 ökutæki?

File: libs/application/template-api-modules/src/lib/modules/templates/transport-authority/change-operator-of-vehicle/change-operator-of-vehicle.service.ts
Comment (jonarnarbriem) [inline libs/application/template-api-modules/src/lib/modules/templates/transport-authority/change-operator-of-vehicle/change-operator-of-vehicle.service.ts:1]: Sýnist vera sömu breytingar og á co-owner, svo sömu komment :)

File: libs/application/template-api-modules/src/lib/modules/templates/transport-authority/transfer-of-vehicle-ownership/transfer-of-vehicle-ownership.service.ts
Comment (jonarnarbriem) [inline libs/application/template-api-modules/src/lib/modules/templates/transport-authority/transfer-of-vehicle-ownership/transfer-of-vehicle-ownership.service.ts:1]: Sýnist vera sömu breytingar og á co-owner, svo sömu komment :)

**Review by jonarnarbriem — APPROVED** [review]

**Pull request merged**